feat: community docs implementation

AceTheCreator opened the pull request:
Implemented the first iteration of the community docs section which will contain different docs categories... This can be located under docs&resources in the community path

let me know what your thoughts are

TL;DR the docs are copies of PR https://github.com/asyncapi/community/pull/442 and https://github.com/asyncapi/community/pull/449 just for the sake of docs preview

cc @derberg

AceTheCreator committed:
added the community index file and ref to navbar link

AceTheCreator committed:
added first navbar iteration

AceTheCreator committed:
successfully created first orbit sequece

AceTheCreator committed:
enhanced first orbit

AceTheCreator committed:
created second orbit sequences sequences

AceTheCreator committed:
enhanced second orbit sequence

AceTheCreator committed:
implementing landing header

AceTheCreator committed:
adding header content

AceTheCreator committed:
implemented third orbit with icons

AceTheCreator committed:
inseting the discussion page image

AceTheCreator committed:
completed heading UI

AceTheCreator committed:
added shadow to navbar

AceTheCreator committed:
adding event landing section

AceTheCreator committed:
added the meeting section to landing

AceTheCreator committed:
added advocacy section

AceTheCreator committed:
.

AceTheCreator committed:
finaliziong the landing page flow

AceTheCreator committed:
enhancing the landing page

AceTheCreator committed:
enhancing landing page

AceTheCreator committed:
added more community members

AceTheCreator committed:
completed first iteration

AceTheCreator committed:
testing current homepage nav

AceTheCreator committed:
added existing navbar to maintain consistency

AceTheCreator committed:
enhancing landing page to maintain consistency

AceTheCreator committed:
making UI responsive

AceTheCreator committed:
making landing page responsive

AceTheCreator committed:
made meeting section responsive

AceTheCreator committed:
completed landing page responsiveness

AceTheCreator committed:
making UI responsive

AceTheCreator committed:
.

AceTheCreator committed:
.

AceTheCreator committed:
fixed merge conflict withb master

AceTheCreator committed:
.

AceTheCreator committed:
.

AceTheCreator committed:
made us of existing subscription component

AceTheCreator committed:
closing whitespaces

AceTheCreator committed:
enhanced mobile view

AceTheCreator committed:
.

AceTheCreator committed:
updated package.lock

AceTheCreator committed:
.

AceTheCreator committed:
Merge branch 'asyncapi:master' into community

AceTheCreator committed:
.

AceTheCreator committed:
.

AceTheCreator committed:
.

AceTheCreator committed:
fixed merge conflict

AceTheCreator committed:
implemented docs page

AceTheCreator committed:
rendering the first category

AceTheCreator committed:
generated post json for community docs

AceTheCreator committed:
improvised community layout with classname

AceTheCreator committed:
completed the first iteration of docs section

AceTheCreator committed:
made docs category file system based

AceTheCreator committed:
completed the community docs section

AceTheCreator committed:
made docs section title dynamic

AceTheCreator committed:
made docs section responsive

AceTheCreator committed:
added required link

AceTheCreator committed:
.

AceTheCreator committed:
Merge branch 'community' of github.com:AceTheCreator/website-1 into community

AceTheCreator commented:
@magicmatatjahu why's the Netlify preview job not running?

magicmatatjahu commented:
@AceTheCreator We have queue for Netlify builds (max as I remember two in a row), please wait some minutes.

akshatnema commented:
> @magicmatatjahu why's the Netlify preview job not running?

I think from Netlify settings, there has been a restriction to run the jobs only on the `master` branch and not on other branch created in the repository. There could be 2 possible solutions: either change the settings in Netlify for the website repo or you can try deploying your forked repo on your Netlify account and share the deployed link with us. We can preview the PR from that link.

magicmatatjahu commented:
@akshatnema Aaa yes... You are right. @derberg Could you enable build for other branches? I think that you can define which one - in our case it will be production (master) and community.

https://docs.netlify.com/site-deploys/overview/#branch-deploy-controls

akshatnema commented:
> @akshatnema Aaa yes... You are right. @derberg Could you enable build for other branches? I think that you can define which one - in our case it will be production, master and community.

Hmm... I explored and tested the branch deploys right now. The problem exists that even if @derberg change the settings in Netlify, it will still not trigger the branch deploys until and unless a change is being made in the other branch. Since this PR is made before making the changes, there will not be any trigger related to this PR deploy. So make sure, if we change the settings, @AceTheCreator has to close this PR and make another one to trigger the branch deploy.

For a working template, I tested in this PR - https://github.com/akshatnema/website/pull/1 (Yeah :sweat_smile:, AsyncAPI CODEOWNERS workflow also works in my repo, that's why Missy has been tagged for review in PR)

AceTheCreator commented:
> @akshatnema Aaa yes... You are right. @derberg Could you enable build for other branches? I think that you can define which one - in our case it will be production (master) and community.
> 
> https://docs.netlify.com/site-deploys/overview/#branch-deploy-controls

@derberg can you please update Netlify settings to enable build for other branches :)

AceTheCreator committed:
implementing event page

quetzalliwrites commented on components/layout/CommunityDocsLayout.js:
```diff
@@ -0,0 +1,27 @@
+  return (
+    <CommunityLayout
+      title="Community docs and resources"
+      description="Getting started with docs"
```
```suggestion
      description="Getting started with community docs"
```

I think we need to more clearly differentiate that these are a separate set of docs than the spec docs

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+---
+title: 'Become a tsc member'
```
```suggestion
title: 'Become a TSC member'
```

acronyms need to be capitalized 😸

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+---
+# BECOME A  TSC MEMBER!
+
+In this article, you will learn how to become a member of the AsyncAPI Technical Steering Committee (TSC). You will know how to contribute, how to become a committer(maintainer), and realize the benefits of being a TSC member.
```
```suggestion
In this article, you will learn how to become a member of the AsyncAPI Technical Steering Committee (TSC). You will know how to contribute, how to become a committer (maintainer), and the benefits of being a TSC member.
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+The following areas are covered:
+* What is the TSC?
+* Why TSC was formed?
```
```suggestion
* Why was the TSC formed?
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+* What is the TSC?
+* Why TSC was formed?
+* Benefits of becoming a TSC member
+* How to become a TSC member?
```
```suggestion
* How to become a TSC member
```

This thought reads better as a statement :)

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+* Benefits of becoming a TSC member
+* How to become a TSC member?
+
+If you prefer video, watch our [YouTube video](https://www.youtube.com/watch?v=uG_aLF9Z1F0) on how to become a TSC member.
```
```suggestion
If you prefer video, watch our [YouTube video on how to become a TSC member](https://www.youtube.com/watch?v=uG_aLF9Z1F0).
```

For accessibility, the hyperlink should be descriptive of what kind of video this is and suggest where the user will land if they click on this url; merely saying it's a YT vid isn't enough for a screenreader or search engine bots 😄✌🏽

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+* **Maintainers:** Are Contributors who have earned the ability to modify (“commit”) source code, documentation, or other technical artifacts in a project’s repository (“Committers”); They are also referred to as CODEOWNERS on [GitHub](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)
+* **Contributors:** Include anyone in the community that contributes code, documentation, pull request reviews, or other technical artifacts to the Project (“Contributors”);
+
+## What is the Technical Steering Committee?
```
```suggestion
## What is the TSC?
```

Since you **already defined this acronym in line 7** and **all other headers use the acronym** let's keep this consistent. 👍🏽

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+* **Contributors:** Include anyone in the community that contributes code, documentation, pull request reviews, or other technical artifacts to the Project (“Contributors”);
+
+## What is the Technical Steering Committee?
+TSC is a group of maintainers that contribute to the creation and maintenance of AsyncAPI Initiative.
```
```suggestion
The TSC is a group of maintainers that contribute to the creation and maintenance of the AsyncAPI Initiative.
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+## Why the TSC was formed
+The concept of TSC was formed when AsyncAPI joined the Linux Foundation in 2021 to guarantee a system of open governance that promotes ownership and neutrality of all participants in the initiative. 
+
+
```
```suggestion
The concept of the TSC was formed when AsyncAPI joined the Linux Foundation in 2021 to guarantee a system of open governance that promotes ownership and neutrality of all participants in the initiative.
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+ 
+* Recognition from AsyncAPI community
+* Exposure of your portfolio to potential employers
+
```
```suggestion
```

removing miscellaneous space

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+
+## How to become a TSC member
+Any maintainer can be a TSC member. In other words, to become a TSC member, you need to become a maintainer in one of the projects in the AsyncAPI Initiative. 
```
```suggestion
Any maintainer can become a TSC member. In other words, to become a TSC member, you need to become a maintainer in one of the projects of the AsyncAPI Initiative. 
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+## How to become a TSC member
+Any maintainer can be a TSC member. In other words, to become a TSC member, you need to become a maintainer in one of the projects in the AsyncAPI Initiative. 
+
+You can do it by doing at least one of below things in AsyncAPI Initiative GitHub organization:
```
```suggestion
You can become a maintainer by doing at least one of the following items within the AsyncAPI Initiative GitHub organization:
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+* Support and commit to existing projects and be invited by other maintainers to join them as a maintainer
+
+## Make your first contribution
+We know you are nervous to make your first contribution. In this section we guide you on steps to becoming a TSC member:
```
```suggestion
We know you are nervous about making your first contribution. In this section, we guide you on the steps towards becoming a TSC member:
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+1. Pick an issue
+2. Open a pull request
+3. Get it merged
```
```suggestion
3. Get pull request merged
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+1. Pick an issue
+2. Open a pull request
+3. Get it merged
+4. Get invitation to be a code owner
```
```suggestion
4. Receive an invitation to become a code owner
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+Here's how to select an issue to contribute to:
+* Join existing PR reviews
+* Look for issues with [**good first issue** label](https://github.com/issues?page=1&q=is%3Aopen+org%3Aasyncapi+sort%3Aupdated-desc+label%3A%22good+first+issue%22)
```
```suggestion
* Look for issues with a [**good first issue** label](https://github.com/issues?page=1&q=is%3Aopen+org%3Aasyncapi+sort%3Aupdated-desc+label%3A%22good+first+issue%22)
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+Here's how to select an issue to contribute to:
+* Join existing PR reviews
+* Look for issues with [**good first issue** label](https://github.com/issues?page=1&q=is%3Aopen+org%3Aasyncapi+sort%3Aupdated-desc+label%3A%22good+first+issue%22)
+* Join [AsyncAPI Slack community](https://asyncapi.com/slack-invite) and join the `#11_how-to-contribute` channel
```
```suggestion
* Join the [AsyncAPI Slack community workspace](https://asyncapi.com/slack-invite) and the `#11_how-to-contribute` channel
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+* Join existing PR reviews
+* Look for issues with [**good first issue** label](https://github.com/issues?page=1&q=is%3Aopen+org%3Aasyncapi+sort%3Aupdated-desc+label%3A%22good+first+issue%22)
+* Join [AsyncAPI Slack community](https://asyncapi.com/slack-invite) and join the `#11_how-to-contribute` channel
+* Ask maintainers to do a public live stream called **Let's talk about contributing** where they can explain specific part of the project where you want to contribute
```
```suggestion
* Ask maintainers to do a public live stream called **Let's talk about contributing** where they can explain a specific part of the project where you want to contribute
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+**Step 2: Open a pull request**
+
+To open a pull request successfully, follow our [Git workflows](https://github.com/asyncapi/community/blob/master/git-workflow.md) contributors guide.
```
```suggestion
To open a pull request successfully, follow our [Git workflows contributors guide](https://github.com/asyncapi/community/blob/master/git-workflow.md).
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+Use the following steps to open your pull request:
+1. Configure a fork
+2. Work on it locally 
+3. Write an appropriate PR title using [conventional commits specification](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md#conventional-commits)
```
```suggestion
3. Write an appropriate PR title using the [conventional commits specification](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md#conventional-commits)
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+1. Configure a fork
+2. Work on it locally 
+3. Write an appropriate PR title using [conventional commits specification](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md#conventional-commits)
+4. Test PR against Github checks-ensure they all pass
```
```suggestion
4. Test your PR against Github checks to ensure they all pass
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+3. Write an appropriate PR title using [conventional commits specification](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md#conventional-commits)
+4. Test PR against Github checks-ensure they all pass
+
+**Step 3: Get it merged**
```
```suggestion
**Step 3: Get pull request merged**
```

Always avoid using antecedents without defining them clearly in each sentence 😄✌🏽

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+**Step 3: Get it merged**
+
+After making a pull request, allow the maintainers to review your PR. In case no one is alerted to review your PR, feel free to bring the issue to the maintainer's attention by either pinging them on Slack or GitHub. We are sometimes delayed in response for many different reasons, no one intention is to ignore you 🙏🏼 .
```
```suggestion
After making a pull request, allow the maintainers to review your PR. If no one is alerted to review your PR, feel free to bring the issue to the maintainer's attention by pinging them on Slack or GitHub. We sometimes delay in response for many reasons; no one intends to ignore you 🙏🏼 .
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+**Step 4: Invitation to become a maintainer**
+
+After making several contributions to AsyncAPI, maintainers will recognize you. You will then receive an invitation to join the TSC. Look at [this example from the website repository](https://github.com/asyncapi/website/pull/890).
```
```suggestion
After making several contributions to AsyncAPI, maintainers will recognize you. You will then receive an invitation to join the TSC. Look at this [maintainer invitation example from the website repository](https://github.com/asyncapi/website/pull/890).
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+Once you become a maintainer in one of the repositories, you can apply to join a group of [TSC members](https://www.asyncapi.com/community/tsc).
+
+Open a PR in **community** repository to add your name to [the list of members](https://github.com/asyncapi/community/blob/master/TSC_MEMBERS.json). Specify your name, repositories that you maintain and if you are sponsored by some company. For example [see this PR that introduced one of new members](https://github.com/asyncapi/community/pull/277).
```
```suggestion
Open a PR in the **community** repository to add your name to the [list of TSC members](https://github.com/asyncapi/community/blob/master/TSC_MEMBERS.json). Specify your name, repositories that you maintain, and if you are sponsored by some company. For example [see this PR that introduced one of the new TSC members](https://github.com/asyncapi/community/pull/277).
```

quetzalliwrites commented on pages/community/docs/getting-started/become-a-tsc-member.md:
```diff
@@ -0,0 +1,103 @@
+
+Open a PR in **community** repository to add your name to [the list of members](https://github.com/asyncapi/community/blob/master/TSC_MEMBERS.json). Specify your name, repositories that you maintain and if you are sponsored by some company. For example [see this PR that introduced one of new members](https://github.com/asyncapi/community/pull/277).
+
+Joining TSC is something that you are allowed to do as a maintainer by default, as described in [charter signed with Linux Foundation when AsyncAPI joined the foundation](https://github.com/asyncapi/community/blob/master/CHARTER.md). It is not a subject that other TSC members vote on. 
```
```suggestion
Joining the TSC is something that you are allowed to do as a maintainer by default, as described in the [charter signed with the Linux Foundation when AsyncAPI joined the foundation](https://github.com/asyncapi/community/blob/master/CHARTER.md). It is not a subject that other TSC members vote on. 
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+---
+title: 'How to create new repository'
```
```suggestion
title: 'How to create a new repository'
```

quetzalliwrites commented on pages/community/docs/getting-started/index.md:
```diff
@@ -0,0 +1,7 @@
+---
+title: 'Getting started'
+description: 'The essential guide to start using AsyncAPI. Keep it as your reference for all the operations that you can perform on the API resources.'
```
```suggestion
description: 'The essential guide to get started using AsyncAPI. Consider this your reference for all the operations that you can perform on the API resources.'
```

quetzalliwrites commented on pages/community/docs/governance/index.md:
```diff
@@ -0,0 +1,7 @@
+---
+title: 'Governance'
+description: 'Because AsyncAPI is a large, complex tool that touches so much of the IT ecosystem, governance policies are important to meet both internal and external standards.'
```
```suggestion
description: 'Because AsyncAPI is a large, complex tool that touches so much of the IT ecosystem, governance policies are important to meet internal and external standards.'
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+## Configuring the repo
+
+
+## Donating a project to asyncapi
```
```suggestion
## Donating a project to AsyncAPI 
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+6. After getting approval from the TSC, Repo is created by Lukasz or Fran, and the `CI/CD workflow` will be push to the Repo. Next, we invite new maintainers as admins of the repo. 
+ To know more about asyncapi `CI/CD` watch this [video](https://www.youtube.com/watch?v=DsQfmlc3Ubo)
+
+## Github Actions CI/CD in new Repo
```
```suggestion
## Github Actions CI/CD in new repo
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+---
+# How to create new repository
+
+This document is not about creating a repository on GitHub. If this is what you were looking for, then [read GitHub guides on repository creation](https://docs.github.com/en/get-started/quickstart/create-a-repo).
```
```suggestion
This document is not about creating a repository on GitHub. If this is what you were looking for, then [read GitHub's guide on repository creation](https://docs.github.com/en/get-started/quickstart/create-a-repo).
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+This document is not about creating a repository on GitHub. If this is what you were looking for, then [read GitHub guides on repository creation](https://docs.github.com/en/get-started/quickstart/create-a-repo).
+
+This document is about creating a new repository at [AsyncAPI Initiative GitHub organization](https://github.com/asyncapi). This guide explains what is the approval process and default repository configuration options.
```
```suggestion
This document is about creating a new repository in the [AsyncAPI Initiative GitHub organization](https://github.com/asyncapi). This guide explains what is the approval process and default repository configuration options.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+This document is about creating a new repository at [AsyncAPI Initiative GitHub organization](https://github.com/asyncapi). This guide explains what is the approval process and default repository configuration options.
+
+This document is also helpful if you are donating a project to AsyncAPI Initiative, which involves transferring repository ownership, and still becomes its maintainer.
```
```suggestion
This document is also helpful if you donate a project to the AsyncAPI Initiative, which involves transferring repository ownership and becoming its maintainer.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+## Creating a repo for a new project
+
+1. Visit [AsyncAPI Initiative Discussions](https://github.com/orgs/asyncapi/discussions)
```
```suggestion
1. Visit [AsyncAPI Initiative Discussions](https://github.com/orgs/asyncapi/discussions).
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+3. After starting a discussion about your new project for asyncapi, and successfully convienced other members of the community why your project ideas is necessary and you want to maintain the codebase.
+
+
+4. Additionally, when creating proposal specify the initial maintainers of the project and specify if you want to use asyncapi CI/CD workflows or not, so we don't get to ask you. At least 2 code maintainers are needed to start
```
```suggestion
4. Additionally, when creating a proposal, specify the initial maintainers of the project and whether you want to use AsyncAPI CI/CD workflows or not, so we don't need to ask you. At least 2 code maintainers are needed to start.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+
+2. Start a new discussion by clicking on `New discussion` and selecting the `Idea` category.
+Specify in the title that you want to create a new repository. In the description specify what is the purpose of the new repository, and why it should be created under the AsyncAPI organization. See also [example discussion from the past](https://github.com/orgs/asyncapi/discussions/300).
```
```suggestion
2. Start a new discussion by clicking on `New discussion` and selecting the `Idea` category. Specify in the title that you want to create a new repository. In the description, specify what is the purpose of the new repository, and why it should be created under the AsyncAPI organization. See also [example discussion from the past](https://github.com/orgs/asyncapi/discussions/300).
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+Specify in the title that you want to create a new repository. In the description specify what is the purpose of the new repository, and why it should be created under the AsyncAPI organization. See also [example discussion from the past](https://github.com/orgs/asyncapi/discussions/300).
+
+
+3. After starting a discussion about your new project for asyncapi, and successfully convienced other members of the community why your project ideas is necessary and you want to maintain the codebase.
```
```suggestion
3. After starting a discussion about your new project for AsyncAPI and successfully convinced other community members why your project ideas are necessary, you want to maintain the codebase.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+4. Additionally, when creating proposal specify the initial maintainers of the project and specify if you want to use asyncapi CI/CD workflows or not, so we don't get to ask you. At least 2 code maintainers are needed to start
+
+5. Your proposal for a new repository must be approved by Technical Steering Committee (TSC). So make sure that at the end of proposal description you mention `@asyncapi/tsc_members` GitHub team to get TSC members notified.
```
```suggestion
5. Your proposal for a new repository must be approved by the Technical Steering Committee (TSC). Make sure that at the end of the proposal description you mention the `@asyncapi/tsc_members` GitHub team so as notify all TSC members.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+5. Your proposal for a new repository must be approved by Technical Steering Committee (TSC). So make sure that at the end of proposal description you mention `@asyncapi/tsc_members` GitHub team to get TSC members notified.
+
+6. After getting approval from the TSC, Repo is created by Lukasz or Fran, and the `CI/CD workflow` will be push to the Repo. Next, we invite new maintainers as admins of the repo. 
```
```suggestion
6. After getting approval from the TSC, the repo is created by either Lukasz or Fran, and the `CI/CD workflow` will be pushed to the repo. Next, we invite new maintainers as repo admins. 
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+5. Your proposal for a new repository must be approved by Technical Steering Committee (TSC). So make sure that at the end of proposal description you mention `@asyncapi/tsc_members` GitHub team to get TSC members notified.
+
+6. After getting approval from the TSC, Repo is created by Lukasz or Fran, and the `CI/CD workflow` will be push to the Repo. Next, we invite new maintainers as admins of the repo. 
+ To know more about asyncapi `CI/CD` watch this [video](https://www.youtube.com/watch?v=DsQfmlc3Ubo)
```
```suggestion
 To learn more, watch this [AsyncAPI `CI/CD video](https://www.youtube.com/watch?v=DsQfmlc3Ubo)
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+ To know more about asyncapi `CI/CD` watch this [video](https://www.youtube.com/watch?v=DsQfmlc3Ubo)
+
+## Github Actions CI/CD in new Repo
+Using asyncapi `CI/CD` workflow saves you from a lot of work load, after including it in your Github discussion/proposal specifing that you want to use our `CI/CD` workflow, the following activities will be carried out by Lukasz or Fran.
```
```suggestion
Using AsyncAPI `CI/CD` workflow saves you from a lot of work load, after including it in your Github discussion/proposal specifing that you want to use our `CI/CD` workflow, the following activities will be carried out by Lukasz or Fran.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+## Github Actions CI/CD in new Repo
+Using asyncapi `CI/CD` workflow saves you from a lot of work load, after including it in your Github discussion/proposal specifing that you want to use our `CI/CD` workflow, the following activities will be carried out by Lukasz or Fran.
+
+1. Go to asyncapi `/.github` repo
```
```suggestion
1. Go to AsyncAPI `/.github` repo.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+A set of `pull request` is created by the asyncapi bot for a given repo, you need to `merge` them in as a maintainer of the repo.
+
+If you don't want to use asyncapi global workflow, then you will have to automate all the necessary `pull request` that would have been made by bot.
```
```suggestion
If you don't want to use the AsyncAPI global workflow, you will have to automate all the necessary `pull requests` that the bot would have made.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+We open a `pull request` in `.github` repo, select `global_replicator.yml` and add the newly created repo to every single `Job` from the `global_replicator.yml` file. Add as **list of repo to be ignore** in `repos_to_ignore: name-of-repo`
+
+This tells the asyncapi bot not to constantly push `pull request` to the repo.
```
```suggestion
The previous action tells the AsyncAPI bot not constantly push `pull request` to the repo.
```

Always avoid starting or ending a sentence with the antecedent `this` because that usually means you are unclear on what you're referring to. 👍🏽

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+If you don't want to use asyncapi global workflow, then you will have to automate all the necessary `pull request` that would have been made by bot.
+
+We open a `pull request` in `.github` repo, select `global_replicator.yml` and add the newly created repo to every single `Job` from the `global_replicator.yml` file. Add as **list of repo to be ignore** in `repos_to_ignore: name-of-repo`
```
```suggestion
We open a `pull request` in the `.github` repo, select `global_replicator.yml` and add the newly created repo to every single `Job` from the `global_replicator.yml` file. Add as **list of repos to be ignored** in `repos_to_ignore: name-of-repo`
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+We have one `worklfow` that basically takes all the workflows from `.github` and we can selectively push to a specific repo. One of the owners which at this time is Lukasz or Fran becuase they have not transfer ownership to the new maintainers executes this actions.
+
+A set of `pull request` is created by the asyncapi bot for a given repo, you need to `merge` them in as a maintainer of the repo.
```
```suggestion
The AsyncAPI bot creates a set of `pull requests` for a given repo. You need to `merge` them as a repo's maintainer.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+4. Click on `run workflow` and paste the name of the new repo.
+5. `Run workflow`.
+
+We have one `worklfow` that basically takes all the workflows from `.github` and we can selectively push to a specific repo. One of the owners which at this time is Lukasz or Fran becuase they have not transfer ownership to the new maintainers executes this actions.
```
```suggestion
We have a `workflow` that takes all the workflows from `.github` and allows you to push to a specific repo selectively. One of the owners (which at the time of this writing is still Lukasz or Fran) executes these actions.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+Using asyncapi `CI/CD` workflow saves you from a lot of work load, after including it in your Github discussion/proposal specifing that you want to use our `CI/CD` workflow, the following activities will be carried out by Lukasz or Fran.
+
+1. Go to asyncapi `/.github` repo
+2. Click on `Action` to view all workflows
```
```suggestion
2. Click on `Action` to view all workflows.
```

quetzalliwrites commented on pages/community/docs/getting-started/how-to-create-new-repository.md:
```diff
@@ -0,0 +1,54 @@
+
+1. Go to asyncapi `/.github` repo
+2. Click on `Action` to view all workflows
+3. Select `Global workflow to rule then all`
```
```suggestion
3. Select `Global workflow to rule then all`.
```

quetzalliwrites reviewed (changes requested):
Here's the editorial review mate!! 😄✌🏽 @AceTheCreator

derberg commented:
current settings are

![Screenshot 2022-09-21 at 09 12 24](https://user-images.githubusercontent.com/6995927/191438536-ad06db8f-d96e-4f1d-8da0-d060ba27cf49.png)

netlify did not react on PR reopen event, so @AceTheCreator you have to create new PR as @akshatnema wrote I guess

akshatnema commented:
@AceTheCreator try making an empty commit to trigger the deployment else you have to make a new PR.

AceTheCreator committed:
Update index.md

AceTheCreator commented:
The preview is ready, @derberg @magicmatatjahu @alequetzalli  @akshatnema can you kindly share your thought on this :)

akshatnema commented on css/styles.css:
```diff
@@ -0,0 +1,288 @@
+/*! purgecss start ignore */
```
This file should be completely removed because we have moved the CSS of tailwind from v1 to v3, for which we are using `styles/global.css` to add any customize classes. So, kindly make changes to that file only.

akshatnema commented on pages/community/docs/index.js:
```diff
@@ -0,0 +1,75 @@
+    setCategories(newList)
+  },[])
+  return (
+    <CommunityLayout
```
Use `GenericLayout.js` component here to provide basic headers and Nav to the page.

akshatnema commented on pages/community/index.js:
```diff
@@ -5,6 +5,7 @@ import Button from '../../components/buttons/Button';
 import NewsletterSubscribe from '../../components/NewsletterSubscribe';
+import Link from 'next/link';
 
 const orbit1 = [
```
If possible, make these array data stored in another js file, for example like inside a new `utils` folder because having this raw array data in the file makes the file less readable and very long in terms of lines. It's better to have these content in seperate file and we import the content from it.

akshatnema commented on pages/community/index.js:
```diff
@@ -5,6 +5,7 @@ import Button from '../../components/buttons/Button';
 import CommunityLayout from '../../components/layout/CommunityLayout';
 import IconRocket from '../../components/icons/Rocket';
 import NewsletterSubscribe from '../../components/NewsletterSubscribe';
+import Link from 'next/link';
```
Use `components/typography/Textlink.js` here instead of using `next/link` because it will then apply the characteristics and properties applied by the Missy and we will have uniform component for links everywhere in the website. Need to not to worry about server-side rendering here because we already use `next/link` inside `Textlink.js` file to handle it.

akshatnema reviewed (changes requested):
@AceTheCreator tbh, it's great work done by you so far. I'm truly fascinated with the kind of work you have done to make the community page working perfectly for website. I'm here with some of my concerns regarding the changes. This is still not a complete review from my side because I have so many changes and doubts in my mind and will be taking 2-3 reviews to complete this. Here are some concerns I want to highlight:

- **Change CommunityLayout to GenericLayout**
We already have a `GenericLayout` file comprising of the head and Navbar in it which you can use instead of separately making a file for CommunityLayout. 

- **Change in the layout of community docs**
I'm not sure you should have a completely different layout specially for the community docs. Please confirm this design first from @mcturco and the theme you have followed. I would better suggest having a review from @mcturco as well.

- **Complete the changes made by @alequetzalli first**
Please make the changes provided by @alequetzalli first so that we will get confirmation about the documentation and content changed/added in this PR and then we can then go further for reviews in the PR.

mcturco commented:
> Change in the layout of community docs
I'm not sure you should have a completely different layout specially for the community docs. Please confirm this design first from @mcturco and the theme you have followed. I would better suggest having a review from @mcturco as well.

I have to agree with @akshatnema here, we should just use the DocsLayout that we have already for any kind of documentation. No need to re-invent the wheel, but more importantly we don't want to confuse users and offer different ways of reading docs!

quetzalliwrites commented:
@AceTheCreator please accept all my commit suggestions to make it feasible for @Dule-mart to copy your latest version into his own PR.

AceTheCreator commented:
> > Change in the layout of community docs
> > I'm not sure you should have a completely different layout specially for the community docs. Please confirm this design first from @mcturco and the theme you have followed. I would better suggest having a review from @mcturco as well.
> 
> I have to agree with @akshatnema here, we should just use the DocsLayout that we have already for any kind of documentation. No need to re-invent the wheel, but more importantly we don't want to confuse users and offer different ways of reading docs!

The reason I went for this approach is just to keep things simple :) Since the community docs are in categories and some categories only consist of a few docs, then this approach will really make more sense for such scenarios.

AceTheCreator committed:
completed the event header banners

AceTheCreator committed:
.

AceTheCreator committed:
created events posts

AceTheCreator committed:
implement event card UI

derberg commented:
yeah, I agree to stick to current docs layout for community docs as you get many stuff out of the box:
- feedback
- right hand side toc
- consistency 😛

AceTheCreator committed:
added filter functionality

AceTheCreator committed:
added events path UI

AceTheCreator committed:
made page responsive

AceTheCreator committed:
reused blog layout for community docs

AceTheCreator committed:
Merge branch 'asyncapi:community' into community

AceTheCreator committed:
Merge branch 'community' of github.com:AceTheCreator/website-1 into community

AceTheCreator committed:
reused the existing generic layout

AceTheCreator committed:
replaced links

AceTheCreator committed:
removed docs

AceTheCreator committed:
fixed the post duplication issue

AceTheCreator committed:
.

AceTheCreator committed:
removed unneccessary console

AceTheCreator commented:
Updated this PR with the suggested changes made by @akshatnema @derberg and @mcturco 

Added the [events page](https://deploy-preview-950--asyncapi-website.netlify.app/community/events) UI also :) 

Kindly review this PR with fresh eyes

akshatnema reviewed (changes requested):
@AceTheCreator Great work so far, hats off! :clap:. I have left some comments on here for certain changes.

- ### Docs pages not working
![Screenshot from 2022-09-30 13-58-44](https://user-images.githubusercontent.com/76521428/193228492-f68a2285-2914-44a5-892a-6eee692b22ed.png)

The content inside the page is not rendering. Kindly look into it.

- ### Unnecessary extra spacing between components
![image](https://user-images.githubusercontent.com/76521428/193229220-7908cd2f-5666-4e77-a12a-a3b910365ba0.png)

I don't think we should have so much space between these 2 components.

- ### Card's background matches with the website's background
![image](https://user-images.githubusercontent.com/76521428/193230655-cac9f7fd-da47-4c81-8017-0c841d0bd7e0.png)

These cards are linked to different community pages, and the way they are highlighted doesn't specify to users that they are links to other pages. I would suggest you change the background colors of these cards to some interactive ones with some animation on hover to make them look good. It's my opinion and do ask some design contributors so they would suggest better colors and approaches.

- ### Button size is too small as compared to text in the block
![image](https://user-images.githubusercontent.com/76521428/193233890-aef05ed0-9c44-4d34-b9c9-816d56d5a7cd.png)

I think we should have the normal button size as we have on the homepage.

- ### Link type property in cards
![Screenshot from 2022-09-30 14-34-36](https://user-images.githubusercontent.com/76521428/193235543-64eefbda-fc07-42b1-9337-4f5fdcae74df.png)

The cards pointing to some pages should have a cursor pointer on hover.
 
- ### Description font size is very small
![image](https://user-images.githubusercontent.com/76521428/193236521-6eba3ec7-bbe2-41da-b564-c7d59f12a858.png)

I would suggest to kindly have the font-size as normal as we use in Blog page description.

AceTheCreator commented:
> @AceTheCreator Great work so far, hats off! 👏. I have left some comments on here for certain changes.
> 
> * ### Docs pages not working
> 
> ![Screenshot from 2022-09-30 13-58-44](https://user-images.githubusercontent.com/76521428/193228492-f68a2285-2914-44a5-892a-6eee692b22ed.png)
> 
> The content inside the page is not rendering. Kindly look into it.

>
Intentionally removed the contents cuz the reason why added it before was that I was using a custom UI and wanted you, folks, to see what the result looks like :)
> * ### Unnecessary extra spacing between components
> 
> ![image](https://user-images.githubusercontent.com/76521428/193229220-7908cd2f-5666-4e77-a12a-a3b910365ba0.png)
> 
> I don't think we should have so much space between these 2 components.

 You sure about this? Cuz if I didn't add the extra white space, it will look like a bunch of texts jam-packed without segregation

> * ### Card's background matches with the website's background
> 
> ![image](https://user-images.githubusercontent.com/76521428/193230655-cac9f7fd-da47-4c81-8017-0c841d0bd7e0.png)
> 
> These cards are linked to different community pages, and the way they are highlighted doesn't specify to users that they are links to other pages. I would suggest you change the background colors of these cards to some interactive ones with some animation on hover to make them look good. It's my opinion and do ask some design contributors so they would suggest better colors and approaches.
>
Yea, I'm leaving it this way just for the sake of having the MVP. The big plan is to ensure each card is an illustration kinda design with different backgrounds. We'll definitely need a designer to help with this
> * ### Button size is too small as compared to text in the block
> 
> ![image](https://user-images.githubusercontent.com/76521428/193233890-aef05ed0-9c44-4d34-b9c9-816d56d5a7cd.png)
> 
> I think we should have the normal button size as we have on the homepage.
> 
> * ### Link type property in cards
> 
> ![Screenshot from 2022-09-30 14-34-36](https://user-images.githubusercontent.com/76521428/193235543-64eefbda-fc07-42b1-9337-4f5fdcae74df.png)
> 
> The cards pointing to some pages should have a cursor pointer on hover.
> 
> * ### Description font size is very small
> 
> ![image](https://user-images.githubusercontent.com/76521428/193236521-6eba3ec7-bbe2-41da-b564-c7d59f12a858.png)
> 
> I would suggest to kindly have the font-size as normal as we use in Blog page description.

AceTheCreator committed:
made minor changes

AceTheCreator committed:
Merge branch 'asyncapi:community' into community

AceTheCreator commented:
@derberg can we get this PR merged? Need to move to the next section

AceTheCreator commented:
@akshatnema @mcturco @magicmatatjahu please i need this PR approved as soon as possible

magicmatatjahu commented:
@AceTheCreator Why `soon as possible`? 🤔

AceTheCreator commented:
> @AceTheCreator Why `soon as possible`? 🤔

Wanna get rid of my fork and refork it again due to some personal setup issue.

magicmatatjahu commented:
@AceTheCreator ok, I thought you wanna merge that PR to the master. I don't mind fix things in the next PRs, but I see that others have given you things to improve.

AceTheCreator commented:
> @AceTheCreator ok, I thought you wanna merge that PR to the master. I don't mind fix things in the next PRs, but I see that others have given you things to improve.

I want it merged with the community branch and I've also updated it with most of the suggested changes. 

So once it's been merged, i can go ahead and delete my fork and fix the issue I'm facing over here

magicmatatjahu reviewed (approved):
I accept that (please check previous comments from @AceTheCreator) but @mcturco and @akshatnema should also.

akshatnema commented:
> I want it merged with the community branch and I've also updated it with most of the suggested changes.

@AceTheCreator If you want to get this merged with `community` branch right now, I think there is no requirement for final reviews at all. `community` branch is just created for the completion of the community route and its pages. So, we can definitely approve and merge this PR now and then, can later on suggest changes to upcoming PRs.

AceTheCreator commented:
Yo, @alequetzalli can you please approve this PR 🙂

quetzalliwrites commented:
Hey @AceTheCreator I don't think I can approve this PR yet because:
- there are still some commit suggestions you haven't accepted
- the branch has conflicts, which I assume is why it hasn't deployed a preview link (I can't preview the new changes so I can't approve)
😄✌🏽

AceTheCreator commented:
> Hey @AceTheCreator I don't think I can approve this PR yet because:
> 
> * there are still some commit suggestions you haven't accepted
> * the branch has conflicts, which I assume is why it hasn't deployed a preview link (I can't preview the new changes so I can't approve)
>   😄✌🏽

@alequetzalli the commit suggestions are outdated and I resolved the conflict. It also has a preview link https://deploy-preview-950--asyncapi-website.netlify.app/ :)

derberg commented:
@AceTheCreator 👇🏼 😉 

<img width="1458" alt="Screenshot 2022-10-12 at 12 19 02" src="https://user-images.githubusercontent.com/6995927/195317525-ded0cf40-dfc9-4ea6-a260-a0eba1b9b4a3.png">

AceTheCreator committed:
Merge branch 'community' into community

AceTheCreator commented:
> @AceTheCreator 👇🏼 😉
> 
> <img alt="Screenshot 2022-10-12 at 12 19 02" width="1458" src="https://user-images.githubusercontent.com/6995927/195317525-ded0cf40-dfc9-4ea6-a260-a0eba1b9b4a3.png">

Thanks for pointing it out. Can't believe I missed it even with 4 eyes :)

quetzalliwrites commented:
Hmm, I actually see several bugs to address before we can merge this:

1.  The animation is LOVELY but the user will be confused because the mouse behavior makes the user think it's clicackle. This should only happen for the CTA button in that section. **Here's a screen recording showing the issue:**
https://user-images.githubusercontent.com/19964402/196254980-f62f42de-1115-4380-b4d6-64aed9836f3d.mov 

2. The `Ambassador programs` and `Community Members` link in **Home of #CommunityOps** section 404s. Is this PR dependant on another PR that you need to merge together? 🤔 Or is this a bug? 🐛 
<img width="600" alt="Screen Shot 2022-10-17 at 11 33 41 AM" src="https://user-images.githubusercontent.com/19964402/196270854-f3434fc9-aa4b-4494-ad34-32b589ceaf4a.png">

3. There is a typo in the slide for the 22' Conference; `Con` instead of `Conf`.
<img width="600" alt="Screen Shot 2022-10-17 at 12 58 01 PM" src="https://user-images.githubusercontent.com/19964402/196271258-18a1a897-1dd8-4baf-8cbb-986ed4b28625.png">

4. Should the **Ambassador Programs** section link to the `/events` page? Wouldn't it make more sense for this to link to an 'Ambassador' page? 🧐 
<img width="600" alt="Screen Shot 2022-10-17 at 1 01 04 PM" src="https://user-images.githubusercontent.com/19964402/196272377-0f89b210-2d06-4261-bb71-273e97ee8236.png">

5. The 3 cards in the following section are confusing as a user because I cannot engage with them. Seeing those cards, users tend to expect them to be clickable and to take you to the section they mention. As such, I would expect these cards to link to the following 3 spots: **tutorials**, **use cases**, and **roadmap**.
<img width="600" alt="Screen Shot 2022-10-17 at 1 04 52 PM" src="https://user-images.githubusercontent.com/19964402/196272772-db3bf575-4d07-4812-a2f8-48824ce18bc7.png">

6. Why does the meta data title for this **Community** landing page say "Events"? It should be `Community`.
<img width="600" alt="Screen Shot 2022-10-17 at 1 08 06 PM" src="https://user-images.githubusercontent.com/19964402/196273395-ddbe7837-d7b1-4971-9c2c-e69cd9609d6f.png">

7. The `Docs & Resources` section on the new `/community/docs` page is a horrible user experience right now. The other main problem is that it causes confusion with the current Docs. Why is there a different "Getting Started" set of docs here when we already have a quickstart in our DevDocs? 
<img width="600" alt="Screen Shot 2022-10-17 at 1 26 53 PM" src="https://user-images.githubusercontent.com/19964402/196276669-7c76e5d6-23d4-45bd-a2c1-a7840bc7c300.png">

quetzalliwrites committed:
typo fix

quetzalliwrites committed:
fix typo

this was correct, do not remove the dash

quetzalliwrites committed:
typos & wording

quetzalliwrites commented on pages/community/events/index.js:
```diff
@@ -0,0 +1,117 @@
+          </div>
+          <div className="text-white w-full sm:w-[50%] p-5 flex flex-col items-center justify-center h-full">
+            <div>
+              <Heading level="h2" typeStyle="heading-md-semibold">
```
Why is TSC plural? As far as I know, we only have one TSC group, with many members? 🤔

AceTheCreator commented:
@alequetzalli while you gave one hell of a review, thanks so much. 

Concerning your review to the pages either with the wrong link or page doesn't exist, the reason is because I'm currently on the first iteration of this feat. And the first iteration is basically me creating the UI for every page with test functionality. 

So this PR is not the final PR :) 

I only want it to get merged to the community branch because I'm currently having some upstream issue from my local version and I wanna get rid of it and set it up again.

And concerning the docs section, @derberg can you share your opinion on that?

derberg commented:
@AceTheCreator you sure you wanted to close this one?

@alequetzalli not sure if you are aware, discussion was somewhere is a related issue. But anyway, this PR is not directed to `master` but a temporary `community` branch. The reason is that changes are big, and @AceTheCreator and @akshatnema didn't want to push one gigantic PR at once to `master`. So @AceTheCreator is iteratively adding features to `community` branch one by one. I think we should scope our reviews to topics related to the PR, and overall/final review once `master` branch is a target. Does it make sense?
